src/components/TriggerFolderLink.jsx

```diff
@@ -0,0 +1,83 @@
+import React, { Component, PureComponent } from 'react'
+import classNames from 'classnames'
+
+import Icon from 'cozy-ui/react/Icon'
+
+import styles from '../styles/triggerFolderLink'
+
+const ERRORED = 'ERRORED'
+const FETCHING = 'FETCHING'
+const SUCCESS = 'SUCCESS'
+
+/**
+ * Cache
+ */
+let driveUrl
+
+/**
+ * Renders a link only if href prop is provided
+ * @extends PureComponent
+ */
+class SmartLink extends PureComponent {
+  render() {
+    const { className, href } = this.props
+    return href ? (
+      <a className={className} href={href}>
+        {this.props.children}
+      </a>
+    ) : (
+      <span className={className}>{this.props.children}</span>
+    )
+  }
+}
+
+export class TriggerFolderLink extends Component {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      status: FETCHING
+    }
+  }
+
+  async componentDidMount() {
+    if (!driveUrl) {
+      try {
+        const { client } = this.context
+        const { data } = await client.collection('io.cozy.apps').all()
+        const driveApp = data.find(app => app.slug === 'drive')
+        driveUrl = driveApp && `${driveApp.links.related}#/files/`
+      } catch (error) {
+        this.setState({
+          status: ERRORED
+        })
+      }
+    }
+
+    this.setState({
+      status: SUCCESS
+    })
+  }
+
+  render() {
+    const { label, trigger } = this.props
+    const { status } = this.state
+    const disabled = status !== SUCCESS || !driveUrl
+    const fetching = status === FETCHING
+    return (
+      <SmartLink
+        className={classNames(styles['col-account-folder-link'], {
+          'u-silver': disabled,
+          'u-c-wait': disabled && fetching,
+          'u-c-not-allowed': disabled && !fetching
+        })}
+        href={driveUrl && `${driveUrl}${trigger.message.folder_to_save}`}
```

Comment:
Since we're only using `folder_to_save ` from `trigger`, I suggest we pass that as a prop instead of the whole object.

Reply:
I find that component working on whole known (triggers, konnectors) "models" are easier to understand. Otherwise you end up having to invent a lot of prop names that are not really necessary.